src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2132,6 +2142,12 @@ private static PartitionIterator fetchRows(List<SinglePartitionReadCommand> comm
             reads[i].awaitResponses();
         }
 
+        // update sensor values based on replica responses
+        for (int i=0; i<cmdCount; i++)
+        {
+            reads[i].updateSensorValues();
```

Comment:
@sbtourist I address this comment as follows:
* Propagated `RequestSensors` from the request thread to the callback response simply by capture an instance of it when the callback instance is created. Processing writer UCU responses is now done on `ReadcallBack#onResponse`
* Decoupled the `syncAllSensors` call from the `PartitionIterator#doOnClose` because we need to sync all replicas, not quorum. I added an `autoSync` option when incrementing each request. I understand the performance concern, if the approach makes sense I can consider async ways to do it. 

See https://github.com/datastax/cassandra/pull/952/commits/b4362cda6e1a941f7413c2ed556158bd19a88d72